packages/cdktf/lib/terraform-conditions.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import { Expression } from "./tfExpression";
+
+export interface TerraformCondition {
+  /**
+   *  This is a boolean expression that should return true if the intended assumption or guarantee is fulfilled or false if it does not.
+   */
+  readonly condition: Expression;
+  /**
+   * This contains the text that Terraform will include as part of error messages when it detects an unmet condition.
+   */
+  readonly errorMessage: string;
+}
+
+/**
+ * Terraform checks a precondition before evaluating the object it is associated with.
+ */
+export interface PreCondition extends TerraformCondition {}
```

Comment:
consistency nit (or "not so nit", because this is publicly facing API): I'd prefer us to go with `Precondition` (to be in line with `precondition` as the attribute where this is used).

Reply:
Good point 👍